src/rules/enforce-firestore-facade.ts

```
@@ -18,6 +18,31 @@ const isMemberExpression = (
+
+const COLLECTION_CONSTRUCTORS = new Set(['Set', 'Map', 'WeakSet', 'WeakMap']);
+
+const isCollectionObjectAssignment = (node: TSESTree.Node): boolean => {
+  if (node.type !== AST_NODE_TYPES.VariableDeclarator) return false;
+
+  const init = node.init;
+  if (!init) return false;
+
+  // Check for direct collection constructor calls
+  // e.g., const mySet = new Set<string>(); const myMap = new Map<string, any>();
+  if (init.type === AST_NODE_TYPES.NewExpression &&
```

ESLint

[error] Insert `⏎····`

(prettier/prettier)